packages/validation/src/Validity.ts

```diff
@@ -13,15 +13,31 @@ export class Valid {
   }
 
   public static notEmpty(argValue: string): boolean {
-    return (
-      (argValue &&
-        Object.prototype.hasOwnProperty.call(argValue, "length") &&
-        argValue.length > 0) ||
-      Object.keys(argValue).length > 0
+    return !(
+      // null
+      (
+        typeof argValue === "undefined" ||
+        argValue === null ||
+        // string
+        (typeof argValue === "string" && argValue.toString().length === 0) ||
+        // Array
+        (argValue.hasOwnProperty?.("length") && argValue.length === 0) ||
+        // Object
+        (typeof argValue === "object" && Object.keys(argValue).length === 0)
+      )
     );
   }
 
   public static email(email: string): boolean {
+    // do not test if no email is provided (use notEmpty for required fields)
+    if (
```

Comment:
can we have here just if(!email) ?

Reply:
no, since we are testing for validity only if email is provided, so email='' , email = null, email = undefined should pass the test